e2e_test.go

```diff
@@ -118,11 +118,25 @@ func runIntegrationTest(t *testing.T, testFile string) {
 			}
 
 			filePath := filepath.Join(dir, tt.Give)
+			if len(tt.WantDiff) > 0 {
+				args := append([]string{"-d", filePath}, args...)
+				var stderr, stdout bytes.Buffer
+				require.NoError(t, run(args, bytes.NewReader(stdin), &stderr, &stdout),
+					"could not run patch")
+				if stdout.String() == "" {
+					// No matches found when applying patch
+					assert.Equal(t, string(tt.WantDiff), stdout.String())
+				} else {
+					wantSplit := bytes.Split(tt.WantDiff, []byte("\n"))
+					gotSplit := bytes.Split(stdout.Bytes(), []byte("\n"))[2:]
+					assert.Equal(t, wantSplit, gotSplit)
+					assert.Equal(t, tt.WantComment, stderr.Bytes())
```

Comment:
Why are we skipping the first two lines of stdout?
This indicates that there's a bug somewhere maybe?

Also, for the actual equality check, compare the strings, not the bytes.
testify will print a readable string diff when there's a mismatch if you use string,
and it'll print an unreadable hex code diff if you use byte.

So this should just be,

```
assert.Equal(t, string(tt.WantDiff), stdout.String())
assert.Equal(t, string(tt.Wantcomment), stderr.String())
```

And if the assertion requires us to mess with the output too much besides, say, skipping empty lines, there's something wrong with the logic that produces that output.

Reply:
Agree. However, this would require fixing absolute paths to be relative. As we discussed earlier, we can do this next time, as this PR is already big enough.